src/pages/VerdictPage.tsx

```diff
@@ -111,19 +122,65 @@ export default function VerdictPage() {
             problemsObject.getProblem(data.problem_id)?.title ||
             'Custom Submission'
 
-          // Format the date string
-          const dateString = `${month}-${day}-${year}`
+          setProblemName(problemName)
+
+          let langStr = 'unknown'
+          if (data.language === 'cpp') {
+            langStr = 'C++'
+          } else if (data.language === 'java') {
+            langStr = 'Java'
+          } else if (data.language === 'py') {
+            langStr = 'Python'
+          }
+
+          setVerdictNum(data.verdict)
+          const verdictStr = isFinishedJudging
+            ? verdictInfo[data.verdict].description
+            : 'Pending'
+
+          const timeSeconds = data.time
+          const timeMilliseconds = Math.ceil(timeSeconds * 1000)
+          const timeStr = timeMilliseconds + ' ms'
+
+          const memoryKilobytes = data.memory
+          const memoryMegaBytes = Math.ceil(memoryKilobytes / 1000)
+          const memoryStr = memoryMegaBytes + ' MB'
+
+          const code = '```' + data.language + '\n' + atob(data.code)
+
           setCurrentVerdict((prevVerdict: Verdict) => {
             return {
               ...prevVerdict,
-              date: dateString,
+              date: dateStr,
               problem: problemName,
-              verdict: data.verdict,
+              language: langStr,
+              verdict: verdictStr,
+              time: timeStr,
+              memory: memoryStr,
               casesPassed: casesPassedStr,
             }
           })
+
           setTestCases(data.verdict_list)
+          setFileType(data.language)
+          setCode(code)
           setIsLoading(false)
+
+          let finished: boolean = true
+          for (const v of data.verdict_list) {
+            if (v < 3) {
+              finished = false
+              break
+            }
+          }
+
+          setIsFinishedJudging(finished)
+
+          // Stop the interval loop
+          if (finished) {
+            clearInterval(interval)
```

Comment:
This is able to reference the interval variable from the lines below? 🤔

Reply:
yeah hoisting

Reply:
I'll fix this by using a useState.